### PR TITLE
Used DICE evidence in C++ client

### PR DESCRIPTION
**Pull request by ipetr0v**
This PR:
- Updates C++ `AttestationVerifier` to use DICE protos
- Updates C++ transport to use DICE protos

Ref https://github.com/project-oak/oak/issues/3641
Ref https://github.com/project-oak/oak/issues/4074
Ref https://github.com/project-oak/oak/issues/4627

### CODE REVIEW

File: cc/attestation/verification/insecure_attestation_verifier.h [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+#include "absl/status/statusor.h"
+#include "absl/strings/string_view.h"
+#include "cc/attestation/verification/attestation_verifier.h"
+#include "proto/attestation/endorsement.pb.h"
+#include "proto/attestation/evidence.pb.h"
+#include "proto/attestation/verification.pb.h"
+
+namespace oak::attestation::verification {
+
+// Cerifier implementation that doesn't verify attestation evidence and is used for testing.
```
Comment (tiziano88) [inline cc/attestation/verification/insecure_attestation_verifier.h:32]: Verifier

File: cc/client/client.cc [hunk 2]
```diff
@@ -24,38 +24,43 @@
...
+  absl::StatusOr<AttestationResults> attestation_results =
+      verifier.Verify(std::chrono::system_clock::now(), endorsed_evidence->evidence(),
+                      endorsed_evidence->endorsements());
+  if (!attestation_results.ok()) {
+    return attestation_results.status();
   }
 
-  return absl::WrapUnique(new OakClient(
-      std::move(transport), endorsed_evidence->attestation_evidence().encryption_public_key()));
+  return absl::WrapUnique(
```
Comment (tiziano88) [inline cc/client/client.cc:62]: Is it not possible to create it directly behind a unique pointer, instead of wrapping a raw pointer?

**Review by tiziano88 — APPROVED** [review]
Thanks

File: cc/attestation/verification/insecure_attestation_verifier.h (hunk 1 above)
Comment (ipetr0v) [reply cc/attestation/verification/insecure_attestation_verifier.h]: Done

File: cc/client/client.cc (hunk 2 above)
Comment (ipetr0v) [reply cc/client/client.cc:62]: pre-existing: Constructor is currenty marked as private, so `make_unique` won't be able to use it

**Review by k-naliuka — APPROVED** [review]

**Pull request merged**